### ISSUE TITLE
Sellersetu (SA : RET16) - Compliance check

### ISSUE THREAD

**Issue by Jagannath-wits** [ONDC-Official/v1.2.0-logs#1614]
# Sellersetu : RET:16

### On Search Full Catalog Refresh 

- In `/message/catalog/bpp~1providers/0/items/0/@ondc~1org~1return_window` must match format `"duration"`
- In `bpp/providers[0]/items[0]/descriptor**: Code should be provided in `/message/catalog/bpp/providers[0]/items[0]/descriptor`

### On Search Inc Refresh 

- In `/message/catalog/bpp~1providers/0/items/0/quantity/available/count` available count must be either 99 or 0 only
- In `/message/catalog/bpp~1providers/0/items/0/@ondc~1org~1return_window` must match format `"duration"`
- In `/message/catalog/bpp~1providers/0/items/0/@ondc~1org~1time_to_ship` must match format `"duration"`

### On Status Pending 

- In `/start/range/start_time` `"2024-05-07T15:47:49.504Z"` of `on_confirm` mismatched with `/start/range/start_time` `"2024-05-06T15:47:48.940Z"` of `on_status` for Delivery Fulfillment
- In `/start/range/end_time` `"2024-05-07T15:47:49.504Z"` of `on_confirm` mismatched with `/start/range/end_time` `"2024-05-06T16:07:48.940Z"` of `on_status` for Delivery Fulfillment
- In `/end/range/start_time` `"2024-05-09T15:47:49.504Z"` of `on_confirm` mismatched with `/end/range/start_time` `"2024-05-06T16:07:48.940Z"` of `on_status` for Delivery Fulfillment
- In `/end/range/end_time` `"2024-05-09T15:47:49.504Z"` of `on_confirm` mismatched with `/end/range/end_time` `"2024-05-06T16:42:48.940Z"` of `on_status` for Delivery Fulfillment

### On Status Packed 

- Same as above

### On Status Picked 

- Same as above

### On Status Out for Delivery 

- Same as above

### On Status Delivered 

- Same as above

## Flow 3

- Same as above in Flow 2

### On Cancel 

- In `precancel_state.updated_at` of `on_cancel` is not equal with the `on_confirm` `order.updated_at`

### On Cancel 

-  RTO fulfillment end and Delivery fulfillment start location mismatch in `on_cancel`
- The `precancel_state.updated_at` of `on_cancel` is not equal with the `on_status_out_for_delivery` `order.updated_at`

## Flow 6

flow 6 should be provided in a single folder.
For reference, refer the README.md of [ONDC-Official/log-validation-utility](https://github.com/ONDC-Official/log-validation-utility).

@sellersetu-hub

**Comment by aagman945** [comment]
@Jagannath-wits  Can you direct me to resource, which states , which code to use for different categories?

**Comment by Jagannath-wits** [comment]
> @Jagannath-wits Can you direct me to resource, which states , which code to use for different categories?
- @aagman945 
https://docs.google.com/spreadsheets/u/1/d/12A_B-nDtvxyFh_FWDfp85ss2qpb65kZ7/edit#gid=1474472733

**Comment by aagman945** [comment]
@Jagannath-wits Thanks , for this, ` bpp/providers[0]/items[0]/descriptor**: Code should be provided in /message/catalog/bpp/providers[0]/items[0]/descriptor` . I tried HSN number for furnitures in log validation utility , It says you should use GTIN. How can we identify which code type to use for which categories ?

**Comment by Jagannath-wits** [comment]
> @Jagannath-wits Thanks , for this, ` bpp/providers[0]/items[0]/descriptor**: Code should be provided in /message/catalog/bpp/providers[0]/items[0]/descriptor` . I tried HSN number for furnitures in log validation utility , It says you should use GTIN. How can we identify which code type to use for which categories ?

You can use HSN for furnitures.
Will be added to utility soon

**Comment by Ikshan-Tango** [comment]
Hi @Jagannath-wits, 

While consolidating flow 6 we are facing errors from the utility, can you help us in identifying as to why they are happening?

Payload :-
[](http://jsonblob.com/1242004493104439296)
 
 Errors :-
                 "message/order.fulfillments/1": "Mismatch occured while comparing 'Delivery' fulfillment object(without state, tags, instructions) with on_status_pending"

**Comment by Ikshan-Tango** [comment]
https://jsonblob.com/1218845999383830528

Payload ^

**Comment by Ikshan-Tango** [comment]
@Jagannath-wits 

Any heads up on the same?
We're facing same errors from the utility as well as buyer app and it's supposed to be a utility issue (as confirmed through mail via ONDC tech support).

Can we raise a PR meanwhile for all the changes from Flow 1 till Flow 5? We're lagging behind already on the same. Expecting a reply.

Thank you

**Comment by Jagannath-wits** [comment]
> @Jagannath-wits
> 
> Any heads up on the same? We're facing same errors from the utility as well as buyer app and it's supposed to be a utility issue (as confirmed through mail via ONDC tech support).
> 
> Can we raise a PR meanwhile for all the changes from Flow 1 till Flow 5? We're lagging behind already on the same. Expecting a reply.
> 
> Thank you



> @Jagannath-wits
> 
> Any heads up on the same? We're facing same errors from the utility as well as buyer app and it's supposed to be a utility issue (as confirmed through mail via ONDC tech support).
> 
> Can we raise a PR meanwhile for all the changes from Flow 1 till Flow 5? We're lagging behind already on the same. Expecting a reply.
> 
> Thank you



> https://jsonblob.com/1218845999383830528
> 
> Payload ^

This payload is for flow 6 and not 5 
Don't see any errors in utility as of now in on_status calls except 
"missingRouting/Tag": "RoutingTag object is mandatory in Tags of Delivery Object for on_status_picked"
@Ikshan-Tango

**Comment by Ikshan-Tango** [comment]
> This payload is for flow 6 and not 5

That's because we're facing errors in flow 6, hence shared the same payload with you like I've specified in the above message as well. And please refer to POINT 2 for the utility response.

1. Flow 6 erroneous payload - https://jsonblob.com/1218845999383830528
2. Flow 6 utility response - https://jsonblob.com/1242890024197677056
3. It's supposed to be a utility issue (as confirmed through mail via ONDC tech support).
4. Hence may we continue by raising a PR for flows 1 - 5 with the necessary changes implemented? 

Kindly go through the above points and let us know our course of action as soon as possible. 

Thank you
@Jagannath-wits

**Comment by Jagannath-wits** [comment]
> > This payload is for flow 6 and not 5
> 
> That's because we're facing errors in flow 6, hence shared the same payload with you like I've specified in the above message as well. And please refer to POINT 2 for the utility response.
> 
> 1. Flow 6 erroneous payload - https://jsonblob.com/1218845999383830528
> 2. Flow 6 utility response - https://jsonblob.com/1242890024197677056
> 3. It's supposed to be a utility issue (as confirmed through mail via ONDC tech support).
> 4. Hence may we continue by raising a PR for flows 1 - 5 with the necessary changes implemented?
> 
> Kindly go through the above points and let us know our course of action as soon as possible.
> 
> Thank you @Jagannath-wits

Like I said 
> Don't see any errors in utility as of now in on_status calls except
"missingRouting/Tag": "RoutingTag object is mandatory in Tags of Delivery Object for on_status_picked"
- I ran the payload you sent above only
- So please fix that and raise an PR for all flows, make sure the settlement_details and trailing quote_trail is present while submitting flow 6

@Ikshan-Tango

**Comment by Ikshan-Tango** [comment]
@Jagannath-wits pls check the PR #1747

**Comment by sellersetu-hub** [comment]
@Jagannath-wits  PR is ready for review #1747

**Comment by Jagannath-wits** [comment]
# Seller-Setu RET 16

## Flow 1

### on_search_full_catalog_refresh
- **invldAreaCode**: STD code does not match with correct area_code on /on_search i.e. 122001 does not match with std code 011
- **bpp/providers[0]/items[0]/descriptor/code**: code should have 1:EAN as a value in /message/catalog/bpp/providers[0]/items[0]/descriptor/coden (It has been recently updated).
- **prvdr0categories**: Support for variants is mandatory, categories must be present in bpp/providers[0]
- **prvdr1categories**: Support for variants is mandatory, categories must be present in bpp/providers[1]
- **descCode**: short desc_code and long desc_code shouldn't be same. For example: "short_desc": "sellersetu", "long_desc": "sellersetu",
- **images**: images links should be accessible. For example: https://development-ironcore.s3.amazonaws.com/ironcore/main/assets/sellersetu.jpg

### on_search_inc_refresh
- **on_search_inc_refresh_msgId**: Message id should not be same with previous calls as it's incremental push based call

## Flow 2

### select
- **itemProvider[1]**: Item with id 0fae5c4e-407b-428e-b2e1-0edd665e74a9 is not available for provider with id d4665463-7965-414e-a9be-aaf702899e45

### on_select
- **ttstat**: /fulfillments/@ondc/org/TAT (O2D) in /on_select can't be less than @ondc/org/time_ship (O2S) in /on_search
- **priceErr**: Warning: Quoted Price in /on_select INR 2596 does not match with the total price of items in /select INR 1598 i.e price for the item mismatch in on_search and on_select because the item with id 0fae5c4e-407b-428e-b2e1-0edd665e74a9 does not exist in on_search

### on_init
- **message.order**: 'cancellation_terms' in /message/order should not be provided as those are not enabled yet

### on_status_picked
- **missingRouting/Tag**: RoutingTag object is mandatory in Tags of Delivery Object for on_status_picked
- **invoice**: An accessible valid invoice link should be provided

### on_status_out_for_delivery
- **missingRouting/Tag**: RoutingTag object is mandatory in Tags of Delivery Object for on_status_out_for_delivery

### on_status_delivered
- **missingRouting/Tag**: RoutingTag object is mandatory in Tags of Delivery Object for on_status_delivered

## Flow 3

### on_select_out_of_stock
- **ttstat**: /fulfillments/@ondc/org/TAT (O2D) in /on_select can't be less than @ondc/org/time_ship (O2S) in /on_search
- **error.message**: The error.message provided in the on_select_out_of_stock should be in the form of an array with proper error_code and item_id. For example: 
```
"[{\"item_id\":\"I1\",\"error\":\"40002\"},{\"item_id\":\"I2\",\"error\":\"40002\"},{\"item_id\":\"I3\",\"error\":\"40002\"}]"
```

## Flow 4

### on_cancel
- **initiated_by**: The bap_id should be provided instead of bap_uri

## Flow 5

### on_cancel
- **rtoFFObj/end/time**: fulfillment type rto end/time should not be present in /on_cancel when state/desc/code is RTO-Initiated
- **RTO.end.location/id**: RTO fulfillment end location id missing in on_cancel
- **RTO.end.location/DeliveryFulfillment.start/location/id**: RTO fulfillment end and Delivery fulfillment start location id mismatch in on_cancel
- **reason_code**: A valid reason code should be provided which can be triggered by RTO

## Flow 6

### on_status_picked
- **invoice**: An accessible valid invoice link should be provided

### on_update_interim_reverse_qc
- **message/order.payment/@ondc/org/settlement_details/0**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_part_cancel i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type": "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "799.00",
    "settlement_timestamp": "2024-05-30T18:58:26.687Z"
}
```
- **returnFulfillment/id**: Mismatch between the fulfillment Id of on_update_interim_reverse_qc MP2-R1 and fulfillment Id of update_reverse_qc 6658cd72cb4ca7318d1a307a

### on_update_approval
- **message/order.payment/@ondc/org/settlement_details/0**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_part_cancel i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type": "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "799.00",
    "settlement_timestamp": "2024-05-30T18:58:26.687Z"
}
```
- **returnFulfillment/id**: Mismatch between the fulfillment Id of on_update_approval MP2-R3 and fulfillment Id of update_reverse_qc 6658cd72cb4ca7318d1a307a
- **ReturnFulfillment.end**: Return fulfillment end object is missing in on_update_approval
- **ReturnFulfillment.start**: Return fulfillment start object is missing in on_update_approval

### on_update_picked
- **message/order.payment/@ondc/org/settlement_details/0**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_part_cancel i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type": "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "799.00",
    "settlement_timestamp": "2024-05-30T18:58:26.687Z"
}
```
- **returnFulfillment/id**: Mismatch between the fulfillment Id of on_update_picked mp2_sdd_f1-RTO-R2 and fulfillment Id of update_reverse_qc 6658cd72cb4ca7318d1a307a
- **Return.start.location.id**: Return fulfillment start location id is not required in on_update_picked

### on_update_delivered
- **message/order.payment/@ondc/org/settlement_details/0**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_part_cancel i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type": "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "799.00",
    "settlement_timestamp": "2024-05-30T18:58:26.687Z"
}
```
- **message/order.payment/@ondc/org/settlement_details/1**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_reverse_qc i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type":

 "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "399.00",
    "settlement_timestamp": "2024-05-30T19:20:35.155Z"
}
```
- **context/timestamp/**: context/timestamp of /on_update_delivered should be greater than context/timestamp of /update_settlement_reverse_qc
- **returnFulfillment/id**: Mismatch between the fulfillment Id of on_update_delivered mp2_sdd_f1-RTO-R2 and fulfillment Id of update_reverse_qc 6658cd72cb4ca7318d1a307a
- **returnFF/end/time/timestamp**: end/time/timestamp of return fulfillment is missing
- **Return.start.location.id**: Return fulfillment start location id is not required in on_update_delivered

### on_update_interim_liquidated
- **message/order.payment/@ondc/org/settlement_details/0**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_part_cancel i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type": "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "799.00",
    "settlement_timestamp": "2024-05-30T18:58:26.687Z"
}
```
- **message/order.payment/@ondc/org/settlement_details/1**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_reverse_qc i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type": "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "399.00",
    "settlement_timestamp": "2024-05-30T19:20:35.155Z"
}
```
- **returnFulfillment/id**: Mismatch between the fulfillment Id of on_update_interim_liquidated mp2_sdd_f1-RTO-R2 and fulfillment Id of update_liquidated 6658d40dcb4ca7318d1a3390

### on_update_liquidated
- **message/order.payment/@ondc/org/settlement_details/0**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_part_cancel i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type": "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "799.00",
    "settlement_timestamp": "2024-05-30T18:58:26.687Z"
}
```
- **message/order.payment/@ondc/org/settlement_details/1**: Missing payment/@ondc/org/settlement_details as compared to update_settlement_reverse_qc i.e. 
```json
{
    "bank_name": "ABC BANK",
    "beneficiary_name": "SellerSetu Pvt. Ltd",
    "branch_name": "XYZ",
    "settlement_bank_account_no": "123654789",
    "settlement_counterparty": "seller-app",
    "settlement_ifsc_code": "123654789",
    "settlement_phase": "sale-amount",
    "settlement_type": "upi",
    "upi_address": "xyz@abc",
    "settlement_amount": "399.00",
    "settlement_timestamp": "2024-05-30T19:20:35.155Z"
}
```
- **returnFulfillment/id**: Mismatch between the fulfillment Id of on_update_liquidated MP2-R3 and fulfillment Id of update_liquidated 6658d40dcb4ca7318d1a3390

### update_settlement_liquidated
- **context/timestamp/**: context/timestamp of /update_settlement_liquidated should be greater than context/timestamp of /on_update_liquidated

@sellersetu-hub

**Comment by Ikshan-Tango** [comment]
@Jagannath-wits in Flow 3 on_select out of stock error is already in the correct form, can you check it once again?

**Comment by Ikshan-Tango** [comment]
For flow 5, can you send the list for the valid reason codes of RTO? 
LSP side cancellation reason ID is provided which shouldn't be wrong for this case.

**Comment by Ikshan-Tango** [comment]
@Jagannath-wits expecting a reply here.

Kindly answer the following :-
1.  3 on_select out of stock error is already in the correct form, can you point what exactly is wrong in it?
2. valid reason code was provided in flow 5, 018 is of LSP side cancellation.
3. what exactly is wrong with all the payment_detail errors? Can you elaborate on it?

**Comment by Jagannath-wits** [comment]
> @Jagannath-wits expecting a reply here.
> 
> Kindly answer the following :-
> 
> 1. 3 on_select out of stock error is already in the correct form, can you point what exactly is wrong in it?
> 2. valid reason code was provided in flow 5, 018 is of LSP side cancellation.
> 3. what exactly is wrong with all the payment_detail errors? Can you elaborate on it?

- Refer to this [sheet](https://docs.google.com/spreadsheets/d/1_qAtG6Bu2we3AP6OpXr4GVP3X-32v2xNRNSYQhhR6kA/edit#gid=1095388031) for more understanding regarding the reason codes.
- Payment details of the previous calls are not being carried forward, which is mentioned clearly in the above calls.
- You are providing extra `"` in the `on_select_oos`:

"[\"{\"item_id\": \"0fae5c4e-407b-428e-b2e1-0edd665e74a9\", \"error\": \"40002\"}\"]"

**Comment by Ikshan-Tango** [comment]
@Jagannath-wits 
raised PR for compliance check #1801, please review and provide feedback. 
Thank you!

**Comment by sellersetu-hub** [comment]
@Jagannath-wits Please review and provide feedback.

**Comment by Jagannath-wits** [comment]
# Sellersetu

## Flow 1
### On_search_full_catalog
- Variants not captured correctly. There is only one item with variant in it so there should be at least 2 items with some difference in defined variants in it. eg: variant with id 80ee6a72-a1a2-407b-9afc-e80720239614 should have 2 items(armchair) with different amount of units for it to have variants otherwise its just normal item.
- Ean code length incorrect should be either 8 or 13 in length

## Flow 2
### on_select
- Missing object:
 "quantity":
  {
    "available":
    {
      "count": "99"
    },
    "maximum":
    {
      "count": "99"
    }
### on_status
- This is not required for delivery fulfillment and only required if fulfillment.type="Buyer-Delivery", indicates whether fulfillment is ready to ship.
  "code": "state",
  "list": [
    {
      "code": "ready_to_ship",
      "value": "no"
    }
  ]
- fulfillment_delay is also not required when there is no delay and in case of delay, please refer to contract for delay reason codes and format.

## Flow 3
### on_select_oos
- Missing object:
 "quantity":
  {
    "available":
    {
      "count": "0"
    },
    "maximum":
    {
      "count": "0"
    }

## Flow 4
### on_cancel
- delivery fee cant be charged as cancellation occurs preshipment

## Flow 5
### on_cancel_RTO
- Same as above

## Flow 6
### update_settlement_part_cancel
- why did tax come in quote trail when its not present in on_confirm ?

### on_status
- "message/order.fulfillments/0/@ondc/org/TAT": "Mismatch occurred while comparing 'Delivery' fulfillment object with on_confirm on key '@ondc/org/TAT'",
- "message/order.fulfillments/0/start/location":"Mismatch occurred while comparing 'Delivery' fulfillment object with on_confirm on key 'start/location/address' how is new information being added in on_status calls. Eg:street”,
- "message/order.fulfillments/0/start/time":"Mismatch occurred while comparing 'Delivery' fulfillment object with on_confirm on key 'start/time’; Invalid attribute start.time.duration. why is it there?”,
- "message/order.fulfillments/0/start/contact":"Mismatch occurred while comparing 'Delivery' fulfillment object with on_confirm on key 'start/contact’; How did the phone number change ?
- “Invoice”: Why is invoice still not publically accessible? Make sure it also updates after part-cancel.

# How are the errors in on-status different in flow 2 and 6 ?

### on_update_interim_reverse_qc
- "returnFulfillment": "Return fulfillment is missing in on_update_interim_reverse_qc”

### on_update_approval
- "returnFF/start/time/range/start": "start/time/range/start time of return fulfillment should be greater than context/timestamp of on_update_approval

### on_update_delivered
- "returnFF/end/time/timestamp": "returnFF/end/time/timestamp of return fulfillment is missing”

#### Please provide payload and response for each flow in next iteration.
@sellersetu-hub

**Comment by Chaitanya-Keyal** [comment]
@Jagannath-wits 
In Flow 4, since there is no delivery fee, this is what will be sent for quote in on_cancel:
```json
"quote": {
            "breakup": [],
            "ttl": "P1D",
            "price": {
              "currency": "INR",
              "value": "0"
            }
          }
```
Is this correct?

**Comment by Ikshan-Tango** [comment]
> “Invoice”: Why is invoice still not publically accessible? Make sure it also updates after part-cancel.

The url is publically accessible, kindly check on_status_delivered, attaching payload for you reference.
```
"documents": [
                {
                    "url": "https://development-ironcore.s3.amazonaws.com/invoice_path/2024-06-08-555818-Kiraana_Dukaan_SellerSetu.pdf",
                    "label": "Invoice"
                }
            ],

```
@Jagannath-wits

**Comment by Chaitanya-Keyal** [comment]
hey @Jagannath-wits 

we are getting this response for on_cancel(flow 4):
```json
"report": {
            "on_cancel": {
                "invldQuoteTrailPrices": "quote_trail price and item quote price sum for on_cancel should be equal to the price as in on_confirm"
            }
        }
```
and this is the quote we are sending:
```json
"quote": {
                "breakup": [
                    {
                        "@ondc/org/item_id": "f0f8eea0-6506-46f5-8dda-dce3075052d8",
                        "@ondc/org/item_quantity": {
                            "count": 0
                        },
                        "title": "Coffee Table",
                        "@ondc/org/title_type": "item",
                        "price": {
                            "currency": "INR",
                            "value": "0.0"
                        },
                        "item": {
                            "price": {
                                "currency": "INR",
                                "value": "399.0"
                            }
                        }
                    },
                    {
                        "@ondc/org/item_id": "9abf42b4-074b-4300-9b7b-1592c0791802",
                        "@ondc/org/item_quantity": {
                            "count": 0
                        },
                        "title": "Armchair",
                        "@ondc/org/title_type": "item",
                        "price": {
                            "currency": "INR",
                            "value": "0.0"
                        },
                        "item": {
                            "price": {
                                "currency": "INR",
                                "value": "799.0"
                            }
                        }
                    },
                    {
                        "@ondc/org/item_id": "mp2_sdd_f1",
                        "title": "Delivery charges",
                        "@ondc/org/title_type": "delivery",
                        "price": {
                            "currency": "INR",
                            "value": "0.0"
                        }
                    }
                ],
                "ttl": "P1D",
                "price": {
                    "currency": "INR",
                    "value": "0.0"
                }
            }
```

**Comment by Ikshan-Tango** [comment]
> ### Flow 4
> on_cancel
> delivery fee cant be charged as cancellation occurs preshipment
> 

Can you please tell us what should we do in this case? If we remove this delivery fee then we get the error as :-

` "invldQuoteTrailPrices": "quote_trail price and item quote price sum for on_cancel should be equal to the price as in on_confirm",`

This delivery fee is sent by us during on_confirm, since we can't decrease the quote price based on fulfillment delivery charges then this should be included. Please correct me if I'm wrong.

@Jagannath-wits

**Comment by sellersetu-hub** [comment]
@Jagannath-wits  Can you please help us resolve this, its a blocker in generating logs.

**Comment by Jagannath-wits** [comment]
> @Jagannath-wits Can you please help us resolve this, its a blocker in generating logs. Should we ignore this ?

will need to see your logs to confirm

**Comment by sellersetu-hub** [comment]
No haven't added any extra cost in quote. Can we connect over a meet to show you logs.

**Comment by Jagannath-wits** [comment]
> https://development-ironcore.s3.amazonaws.com/invoice_path/2024-06-08-555818-Kiraana_Dukaan_SellerSetu.pdf

Sure https://meet.google.com/rgk-duyi-jqd

**Comment by Jagannath-wits** [comment]
> No haven't added any extra cost in quote. Can we connect over a meet to show you logs.

Sure, https://meet.google.com/rgk-duyi-jqd

**Comment by Jagannath-wits** [comment]
> > No haven't added any extra cost in quote. Can we connect over a meet to show you logs.
> 
> Sure, https://meet.google.com/rgk-duyi-jqd

### on_init
- Only @ondc/org/buyer_app_finder_fee_type, @ondc/org/buyer_app_finder_fee_amount, @ondc/org/settlement_details are required in payment object as only buyer is collecting payment currently. In case seller wants to collect the payment then you can send it the way you sent but then it won't change in confirm call; right now payment is only accepted by buyer on network so the payment object in confirm is correct

**Comment by Ikshan-Tango** [comment]
@Jagannath-wits 
pls review PR #1875

**Comment by sellersetu-hub** [comment]
@Jagannath-wits Please review and provide feedback.

**Comment by Jagannath-wits** [comment]
## Flow 1

### on_search
- **image urls**: Image URLs should be publicly accessible. *(Repeated issue)*
- **item with id 01b398e8-1382-439f-bf6a-eba2db6297dc**: Sells for 96 units of table for 1500rs? Please provide realistic values.

### inc_on_search
- How can `descriptor.code` of items change from EAN to HSN?

## Flow 2

### on_status
- `fulfillment_delay` is only required when there is a delay in delivery. (for every flow)

### on_status_pending
- When the `onConfirm Order State` is 'Accepted', the `on_status_pending` is not required! *(Repeated issue)*

### on_status_picked
- Invoice URL should be publicly accessible and accurate. *(Repeated issue)*

## Flow 3

### on_confirm
- `delivery_fulfillment.start.address`, Gurgaon comes in Haryana state and not in Delhi.

## Flow 6

### on_update_interim_return
- Return fulfillment not captured correctly. Please refer to contract as it doesn't have `start` and `end` in interim calls. *(Repeated issue)*
- Invalid attribute `start.time.duration`. Why is it there? *(Repeated issue)*

### on_update_interim_liquidated
- Return fulfillment not captured correctly. Please refer to contract as it doesn't have `start` and `end` in interim calls. *(Repeated issue)*
- Invalid attribute `start.time.duration`. Why is it there? *(Repeated issue)*

@sellersetu-hub

**Comment by Ikshan-Tango** [comment]
@Jagannath-wits 

> item with id 01b398e8-1382-439f-bf6a-eba2db6297dc: Sells for 96 units of table for 1500rs? Please provide realistic values.

I believe that's not necessary for us in pre-prod as we're using testing DB, in production there will be all realistic values. This error is not a log validation error hence ignoring this.

**Comment by Chaitanya-Keyal** [comment]
@Jagannath-wits 
Since there are no issues with flows 4 and 5, is it still necessary to provide new logs for them?

**Comment by Jagannath-Padhy** [comment]
> @Jagannath-wits Since there are no issues with flows 4 and 5, is it still necessary to provide new logs for them?

Yes it's required for a final check also so please provide the flow 4 and 5 as well

**Comment by aagman945** [comment]
@Jagannath-wits Please review .

**Comment by Jagannath-wits** [comment]
> @Jagannath-wits Please review .

Will update today

**Comment by Ikshan-Tango** [comment]
@Jagannath-wits waiting on this

**Comment by Jagannath-wits** [comment]
# Sellersetu

## Flow 1
#### on_search 
- attribute.brand and material will be in lower case and not upper case. Please refer to contract page no.56

## Flow 2
### init
- `order.items.location_id` is not required as per contract.

### on_init
- `Tags.bpp_terms`: Only `tax_number` and `provider_tax_number` are required; others are not enabled yet. 

## Flow 5
- on_status_rto_delivered is required after on_cancel_RTO where the end.timestamp of RTO fulfillment will be updated.

## Flow 6
#### on_update
- invoice not carried forward, please refer to contract.

@sellersetu-hub

**Comment by Chaitanya-Keyal** [comment]
@Jagannath-wits Please review and provide feedback

**Comment by sandeepshahi** [comment]
#### order flow must be provided with a feasible customer location

## Flow 1

**/on_search**

- range should be used instead of schedule/frequency and schedule/times
- attribute codes should be in lowercase; check attributes like "brand", "material" etc.
- EAN codes provided for SKUs seem to be incorrect; pls check
- store opening days should be provided in sequence
- colour_name attribute is mandatory for furniture category

**/on_search (inc)**

- entire catalog should not be provided in incremental push;check the contract for the scenarios of incremental refresh
- delta changes must be provided in incremental calls

## Flow 2

**/on_init**

- only np_type, tax_number and provider_tax_number are required; other tncs are not enabled

**/on_confirm**

- delivery TAT should not change; must remain the same as in /on_select and is applicable for all the APIs

**/on_status**

- estimated pickup/delivery time ranges must be provided
- tags/state is not required for seller side delivery
- OTP is not used correctly; use "authorization" object if otp is to be used

## Flow 3

- similar issues as above
- invalid Flow; order must be continued with items left in the cart with the same txn id

## Flow 4

- similar issues as above

## Flow 5

- similar issues as above

**/on_cancel**

- pickup timestamp should not be removed from cancelled forward shipment; audit trail must be maintained
- invalid value of "initiated_by"; should be the subscriber_id of entity initiating the RTO
- invalid attribute id for fulfillments/start/location for RTO shipment
- invalid attribute descriptor/name for fulfillments/start/location for RTO shipment; check start and end details for other such issues
- RTO initiation timestamp can't be future dated

**/on_status**

- terminal state for rto (delivered/disposed) should be provided for verification

## Flow 6

- similar issues as above

**/on_update (part cancel)**

- invalid quantity of items in /items; check the total item quantity after part cancellation
- invalid line item for tax in quote_trail
- order/updated_at timestamp doesn't seem to be correctly captured

**/on_update (interim)**

- quote and quote_trail should not be updated on interim state
- order/updated_at is not updated correctly; check in other APIs as well

**/on_update (approval)**

- /items, quote and quote_trail should not be updated on interim state;
- order/updated_at is not updated correctly; check in other APIs as well

**/on_update (picked)**

- /items is not updated correctly
- estimated return pickup time ranges must be feasible and should remain the same as in /on_update (approval)
- invalid line item for tax in quote_trail

**/on_update (liquidated)**

- similar issues as above; audit trail must be correctly maintained


@Ikshan-Tango @Chaitanya-Keyal @aagman945

**Comment by aagman945** [comment]
@sandeepshahi @Jagannath-Padhy  when will order/updated_at get updated, which all cases ?

**Comment by sandeepshahi** [comment]
> @sandeepshahi @Jagannath-Padhy when will order/updated_at get updated, which all cases ?

@aagman945, whenever there is any change in the order object.

**Comment by sellersetu-hub** [comment]
Greetings Team,

@Sandeep Shahi ***@***.***> @***@***.***
***@***.***> It would really mean a lot if you can tell us if
these issues that you pointed out in this iteration are the last ones.

We have solved those issues and will be generating the logs again. It would
be great if you can tell us any other issues that are left or that these
are the last ones so that we can solve them all at once and can convey it
to our clients as well.Every iteration takes about a week and this has been
delaying a lot of big companies that are wanting to onboard ONDC with us.

Again, it would really mean a lot if you can update us on this.

Best Regards
Aagman Bhatt

**Comment by aagman945** [comment]
@sandeepshahi  we have raise the PR https://github.com/ONDC-Official/v1.2.0-logs/pull/1994 , Please review

**Comment by aagman945** [comment]
@sandeepshahi  @Jagannath-Padhy  Please review.

**Comment by Jagannath-wits** [comment]
### Sellersetu

#### Flow 1
item.attributes such as colour_name and material value should be in lowercase as defined in the contract (repeated issue).

#### On_search_inc
Invalid attribute 'extradetail' and do not provide an empty object as the value of attributes. If attribute value is not available, please refrain from providing it.

#### Flow 2
Instructions object not captured correctly. Please refer to the contract, and ensure code and short_desc values are mapped correctly (repeated issue).

#### Flow 6
##### Update_settlement_part_cancel
Invalid payload.

##### On_update_approval
/items, quote, and quote_trail should not be updated in interim state (repeated issue).

@sellersetu-hub @aagmanbhatt

**Comment by sellersetu-hub** [comment]
@Jagannath-wits Please review.

**Comment by sandeepshahi** [comment]
## Flow 1

**/on_search**

- time/timestamp at provider, location and item can't be future dated
- letter casing of attribute enums must be the same as defined [here](https://github.com/ONDC-Official/protocol-network-extension/tree/main/enums/retail)
- invalid "parent_item_id" defined for several items; should be the id of a valid variant group

## Flow 2

**/on_init**

- prepaid payment collection by BPP is not enabled on the network; no need to provide uri (P/G link), type, collected_by, status here

**/on_confirm**

- timestamp in estimated pickup/delivery ranges can't be earlier than context/timestamp; check start/time/range/start

## Flow 4

**/on_cancel**

- fulfillment TAT should not change for cancelled forward shipment

## Flow 5

**/on_cancel**

- similar issues as above

**/on_status (RTO)**

- invalid attribute tags in /order
- invalid tag return_request
- tags with cancel_request, precancel_state missing from forward shipment

## Flow 6

**/on_update (part cancel)**

- fulfillment id for part cancel (seller side) and full cancellation (buyer side) should not be same; what happens in case buyer cancels the full order after receiving the part cancellation request?
- fulfillment TAT should not change

**/on_status**

- item quantities are not mapped correctly in /items; check part cancelled item




@sellersetu-hub @aagmanbhatt

**Comment by sellersetu-hub** [comment]
> timestamp in estimated pickup/delivery ranges can't be earlier than context/timestamp; check start/time/range/start

This start/time/range object is being sent by logistics provider we are directly using that.

> fulfillment id for part cancel (seller side) and full cancellation (buyer side) should not be same; what happens in case buyer cancels the full order after receiving the part cancellation request?

Then second cancellation will have incremental suffice. i.e C1 , C2 and so on 

@sandeepshahi

**Comment by sellersetu-hub** [comment]
@sandeepshahi  We have raised PR, #2103

**Comment by sandeepshahi** [comment]
## Flow 1

**/on_search**

- letter casing of attribute enums must be the same as defined [here](https://github.com/ONDC-Official/protocol-network-extension/tree/main/enums/retail)
- no variants linked with variant_group: "c95011c6-89b5-482f-9ed8-ac0eae9cdea9"; why?
- time/timestamp at provider, location and item not captured correctly; discrepancies in the timestamp provided in full and incremental call

## Flow 4

**/on_cancel**

- duplicate tag codes "cancel_request" and "precancel_state" in /fulfillments/tags

## Flow 5

**/on_cancel**

- why is the there discrepancy in the info of RTO initiation in /fulfillments/tags/cancel_request/initiated_by and /cancellation/cancelled_by

**/on_status (RTO)**

- similar issues as above
- order related info must remain the same as in /on_cancel RTO; audit trail must be maintained with updation of relevant timestamps and status info


@sellersetu-hub

**Comment by sellersetu-hub** [comment]
Hey @sandeepshahi Flow 1 issues was caused by database migrations, We'll regenerate logs. Can we submit logs for only flow 1 ,4 and 5 . Since Issue are in only these.

**Comment by sandeepshahi** [comment]
@sellersetu-hub, yes

**Comment by sandeepshahi** [comment]
@sellersetu-hub, mocked requests should not be provided for verification; Please test with a counterparty NP before submitting the logs

**Comment by sandeepshahi** [comment]
check the timestamp of /search and /on_search

**Comment by sellersetu-hub** [comment]
@sandeepshahi This isn't mocked, we discovered this issue on Saturday( we are creating the response payload using request payload and changing necessary information like timestamp and adding bpp_id and uri) , and has been fixed.

**Comment by sellersetu-hub** [comment]
@sandeepshahi I am creating a PR with latest search request, so i need to submit incremental too?

**Comment by sellersetu-hub** [comment]
Hi @sandeepshahi,

I’ve created a new PR and would really appreciate it if you could review it at your earliest convenience. Your prompt feedback would be highly valued.

**Comment by sandeepshahi** [comment]
## Flow 1

**/on_search**

- time/timestamp at provider, location and item not captured correctly; discrepancies in the timestamp provided in full and incremental call

## Flow 4

**/on_cancel**

- why are pickup and delivery instructions provided when the order is cancelled? New issues should not be created while fixing others


@sellersetu-hub

**Comment by sellersetu-hub** [comment]
@sandeepshahi  can you explain whats wrong in on_search, since inc_on_search is older than on_search, there will be discrepencies ?

**Comment by sellersetu-hub** [comment]
- In inc_on_search there is no location obj, so how is there discrepancy in timestamp
- Provider timestamp is same in both payloads.

**Comment by sellersetu-hub** [comment]
@sandeepshahi  Can you close this issue since log verification is already done?

**Comment by FarheenNazQA** [comment]
## Flow-1

### Search
- Static terms are not enabled yet and hence not required.(message/intent/tags[bap_terms])`

### On_Search
- Return window shouldn't be zero, as the item is returnable. (It also serves an igm window.)
- Categories should be present.

### On_Search_inc
- There should be some delta change in Increment apart from the timestamp.

## Flow-2

### On_confirm 
- Time to ship should be within the `fulfilment.start.range`.

### On_status_packed
- Tracking object is missing for `tracking-true`.
- Pickup time range should match with the time to ship.

### On_status_picked
- Invoice should be publicly accessible.

## Flow-5

### On_cancel_rto
- In on_cancel_rto, RTO `fulfillment.end.time` should not be present. (End timestamp is provided in `on_status_rto_delivered` or disposed, and not before that for RTO fulfillment.)











@Chaitanya-Keyal

**Comment by sellersetu-hub** [comment]
@FarheenNazQA  the logs summitted were for buyer app this is old issue for sellerapp side. @sandeepshahi please look into this
